save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -34,280 +36,192 @@ import react.dom.html.ReactHTML.small
-
-    /**
-     * true if the device is mobile (screen is less 1000px)
-     */
-    var isMobileScreen: Boolean?
+private val topBarUrlSplits: FC<TopBarPropsWithLocation> = FC { props ->
+    nav {
+        className = ClassName("navbar-nav mr-auto w-100")
+        ariaLabel = "breadcrumb"
+        ol {
+            className = ClassName("breadcrumb mb-0")
+            li {
+                className = ClassName("breadcrumb-item")
+                ariaCurrent = "page".unsafeCast<AriaCurrent>()
+                a {
+                    href = "#/"
+                    // if we are on welcome page right now - need to highlight SAVE in menu
+                    val textColor = if (props.location.pathname == "/") "text-warning" else "text-light"
+                    className = ClassName(textColor)
+                    +"SAVE"
+                }
+            }
+            props.location.pathname
+                .substringBeforeLast("?")
+                .split(URL_PATH_DELIMITER)
+                .filterNot { it.isBlank() }
+                .apply {
+                    val url = TopBarUrl(props.location.pathname.substringBeforeLast("?"))
+                    forEachIndexed { index: Int, pathPart: String ->
+                        url.changeUrlBeforeButton(pathPart)
+                        if (url.isCreateButton(index)) {
+                            li {
+                                className = ClassName("breadcrumb-item")
+                                ariaCurrent = "page".unsafeCast<AriaCurrent>()
+                                if (index == size - 1) {
+                                    a {
+                                        className = ClassName("text-warning")
+                                        +pathPart
+                                    }
+                                } else {
+                                    a {
+                                        href = url.currentPath
+                                        className = ClassName("text-light")
+                                        +pathPart
+                                    }
+                                }
+                            }
+                        }
+                        url.changeUrlAfterButton(pathPart)
+                    }
+                }
+        }
+    }
 }
 
-private fun ChildrenBuilder.dropdownEntry(
-    faIcon: dynamic,
-    text: String,
-    handler: ChildrenBuilder.(ButtonHTMLAttributes<HTMLButtonElement>) -> Unit = { },
-) = button {
-    type = ButtonType.button
-    className = ClassName("btn btn-no-outline dropdown-item rounded-0 shadow-none")
-    fontAwesomeIcon(icon = faIcon) {
-        it.className = "fas fa-sm fa-fw mr-2 text-gray-400"
+/**
+ * Displays the static links that do not depend on the url
+ */
+@Suppress("MAGIC_NUMBER")
+private val topBarLinks: FC<TopBarPropsWithLocation> = FC { props ->
+    val linkToSaveCloudOnGithub = "https://github.com/saveourtool/save-cloud"
```

Comment:
I suggest to put it into constants (I think we have some sort of file with constants in frontend?)

Reply:
Done, added SAVE_CLOUD_GITHUB